File: Lecture001 Introduction, Flowchart, PseudoCode, Why Programming/Flowchart Questions Code in C++/leapYear.cpp
#include <bits/stdc++.h>
using namespace std;

bool checkLeapYear(int year)
{
    return (year % 4 == 0 && (year % 100 != 0 || year % 400 == 0));
}
int main()
{
    int testCases;
    cin >> testCases;
    for (int i = 0; i < testCases; i++)
    {
        int year;
        cin >> year;
        if (checkLeapYear(year))
        {
            cout << year << " is a leap year!" << endl;
        }
        else
        {
            cout << year << " is a not leap year!" << endl;
        }
    }
    return 0;
}

File: Lecture001 Introduction, Flowchart, PseudoCode, Why Programming/Flowchart Questions Code in Java/leapYear.java
import java.util.Scanner;

public class leapYear {
    public static boolean checkLeapYear(int year) {
        return (year % 4 == 0 && (year % 100 != 0 || year % 400 == 0));
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int testCases = sc.nextInt();

        for (int i = 0; i < testCases; i++) {
            int year = sc.nextInt();
            if (checkLeapYear(year)) {
                System.out.println(year + " is a leap year!");
            } else {
                System.out.println(year + " is not a leap year!");
            }
        }

        sc.close();
    }
}
